test/parachain/tests/integration_test.rs

```diff
@@ -329,21 +329,72 @@ async fn integration_test() {
 				"--bootnodes=/ip4/127.0.0.1/tcp/27116/p2p/{}",
 				polkadot_bob_id
 			))
+			.arg("--charlie")
 			.spawn()
 			.unwrap();
-		let cumulus_helper = ChildHelper::new("cumulus", &mut cumulus);
+		let cumulus_charlie_helper = ChildHelper::new("cumulus-charlie", &mut cumulus_charlie);
 		wait_for_tcp("127.0.0.1:27017").await;
 
 		// connect rpc client to cumulus
-		let transport_client_cumulus =
+		let transport_client_cumulus_charlie =
 			jsonrpsee::transport::http::HttpTransportClient::new("http://127.0.0.1:27017");
-		let mut client_cumulus = jsonrpsee::raw::RawClient::new(transport_client_cumulus);
+		let mut client_cumulus_charlie = jsonrpsee::raw::RawClient::new(transport_client_cumulus_charlie);
 
 		// wait for parachain blocks to be produced
 		let number_of_blocks = 4;
 		let mut previous_blocks = HashSet::with_capacity(number_of_blocks);
 		loop {
-			let current_block_hash = Chain::block_hash(&mut client_cumulus, None)
+			let current_block_hash = Chain::block_hash(&mut client_cumulus_charlie, None)
+				.await
+				.unwrap()
+				.unwrap();
+
+			if previous_blocks.insert(current_block_hash) {
+				eprintln!("new parachain block: {}", current_block_hash);
+
+				if previous_blocks.len() == number_of_blocks {
+					break;
+				}
+			}
+
+			sleep(Duration::from_secs(2)).await;
+		}
+
+		// run cumulus dave
+		let cumulus_dave_dir = tempdir().unwrap();
+		let mut cumulus_dave = Command::new(cargo_bin("cumulus-test-parachain-collator"))
```

Comment:
What does this test?

Reply:
As I said in https://github.com/paritytech/cumulus/pull/96#discussion_r426663407

I also added a test in the integration test but it is not super useful, it will only detect something if the implementation is broken but not if the implementation returns always Success like the default implementation of BlockAnnounceValidator.

So for example if the implementation is broken and no block at all are validated, then we would see on Dave that no block are being imported and the test would fail.